samwell/sam/sambuilder.py

```diff
@@ -114,14 +117,29 @@ def __init__(self,
             extra_header: a dictionary of extra values to add to the header, None otherwise.  See
                           `::class::~pysam.AlignmentHeader` for more details.
             seed: a seed value for random number/string generation
+            sort_order: optional sort order, if `None` reads will be output in the same order as
+                they were appended. If `SamOrder.Coordinate`, reads will be ordered by reference
+                index and coordinate order. If `SamOrder.QueryName`, reads will be ordered by
+                query name.
         """
 
         self.r1_len: int = r1_len if r1_len is not None else self.DEFAULT_R1_LENGTH
         self.r2_len: int = r2_len if r2_len is not None else self.DEFAULT_R2_LENGTH
         self.base_quality: int = base_quality
         self.mapping_quality: int = mapping_quality
+
+        sort_order = (
+            SamOrder.Unsorted
+            if sort_order is None or sort_order == SamOrder.Unknown
```

Comment:
why have Unknown here?

Reply:
My thought process was that if the sort order should never be `Unknown` when you're creating the SAM / BAM / CRAM yourself via samBuilder. But I guess that's covered by the assert statement on the following line, which is probably preferable.